analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CastShouldNotBeDuplicated.CSharp9.cs

```diff
@@ -13,41 +15,108 @@ class Program
         public void Foo(Object x, Object y)
         {
             var fruit = x as Fruit;
-            if (fruit is not Fruit) // FN
+            if (fruit is not Fruit) // Compliant, redundant condition, not related for the current rule
             {
             }
 
             object o;
-            switch (x)
+            switch (x)                            // Noncompliant [switch-st-0]
```

Comment:
I am not sure the suggestion to use 'as' an null check is a valid one for a switch. So, maybe we should suggest somehow to use the u variable for example in line 32. Or just say to remove the redundant cast.